parsing/builtin_attributes.ml

```diff
@@ -188,3 +188,22 @@ let explicit_arity =
       | ({txt="ocaml.explicit_arity"|"explicit_arity"; _}, _) -> true
       | _ -> false
     )
+
+(* The "ocaml.boxed (default)" and "ocaml.unboxed (default)"
+   attributes cannot be input by the user, they are added by the
+   compiler when applying the default setting. This is done to record
+   in the .cmi the default used by the compiler when compiling the
+   source file because the default can change between compiler
+   invocations. *)
+
```

Comment:
This answers one of my questions in another note.  I would really prefer to have the information be passed properly through a new field in type_declaration since they affect the behavior of the compiler.